src/util/containers/ContainerManager.ts

```diff
@@ -0,0 +1,27 @@
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+
+/**
+ * Handles everything related to container identifiers and how they relate to other identifiers.
+ */
+export interface ContainerManager {
+  /**
+   * Verifies if this identifier could be valid.
+   * This does not check if this identifier actually exists,
+   * but checks if the identifier is something that could be a valid identifier.
+   */
+  isValidIdentifier: (identifier: ResourceIdentifier) => boolean;
+
+  /**
+   * Generates the identifier of the container this resource would be a member of.
+   * This does not check if that identifier actually exists, this is simply string manipulation.
+   * Output is not defined if the input is a root container or an invalid identifier,
```

Comment:
Can't we be consistent for root container and throw?

Reply:
Perhaps same for invalid actually.

Reply:
True, I wanted to allow the option for the user to choose if those functions needed to be called but that's probably too much of a micro optimization.